db/hybrid_logical_vector.go

```diff
@@ -85,7 +85,17 @@ func (hlv *HybridLogicalVector) AddVersion(newVersion CurrentVersionVector) erro
 	if hlv.PreviousVersions == nil {
 		hlv.PreviousVersions = make(map[string]uint64)
 	}
-	hlv.PreviousVersions[hlv.SourceID] = hlv.Version
+	// we need to check if source ID already exists in PV, if so we need to ensure we are only updating with the
+	// sourceID-version pair if incoming version is greater than version already there
+	if currPVVersion := hlv.PreviousVersions[hlv.SourceID]; currPVVersion != 0 {
```

Comment:
I think using the existence check for a map like `if currPVVersion, ok := hlv.PreviousVersions[hlv.SourceID]; ok {` is more idiomatic go for this type of check.